backend/server.py

```diff
@@ -321,64 +347,68 @@ def add_inference():
 def create_fiware_node():
-
-    global fiware_node_content
     fiware_node_content = request.json
-
-    print(f'Creating Fiware node with data: {fiware_node_content}')
-
     global fiware_node
-    fiware_node = FiwareNode(
-        name=fiware_node_content['Name'],
-        server_ip=fiware_node_content['Server IP'],
-        server_port=fiware_node_content['Server Port'],
-        context_broker_ip=fiware_node_content['Context Broker IP'],
-        context_broker_port=fiware_node_content['Context Broker Port'],
-        entity_id=fiware_node_content['Entity ID'],
-        entity_data=fiware_node_content['Attribute Data'],
-        entity_solution=fiware_node_content['Attribute Solution'],
-        domain=fiware_node_content['Domain']
-    )
-
+    try:
+        fiware_node = FiwareNode(
+            name=fiware_node_content['Name'],
+            server_ip=fiware_node_content['Server IP'],
+            server_port=fiware_node_content['Server Port'],
+            context_broker_ip=fiware_node_content['Context Broker IP'],
+            context_broker_port=fiware_node_content['Context Broker Port'],
+            entity_id=fiware_node_content['Entity ID'],
+            entity_data=fiware_node_content['Attribute Data'],
+            entity_solution=fiware_node_content['Attribute Solution'],
+            domain=fiware_node_content['Domain']
+        )
+    except Exception as e:
+        print(f'There has been an error while trying to create the Fiware Node {e}')
+        return jsonify({'Error': f'There has been an error while trying to create the Fiware Node {e}'})
+    
     fiware_run = threading.Thread(target=fiware_node.run)
 
     try:
         fiware_run.start()
-    except:
+        return jsonify({'message': 'OK'})
+    except Exception as e:
         fiware_run.join()
-
-    return jsonify({'message': 'OK'})
+        print(f'There has been an error while trying to start the Fiware Node {e}')
+        return jsonify({'Error': f'There has been an error while trying to start the Fiware Node {e}'})
 
 @app.route('/context_broker/data', methods=['GET', 'POST'])
 def add_data():
-
     content = request.json
 
     global fiware_node
-
+    
     try:
-        fiware_node.post_data(content['data'])
+        fiware_node.post_data(content['data'],default_values.TIMEOUT_POST_DATA)
+        return jsonify({'message': 'OK'})
     except Exception as e:
-        return jsonify({'Error': str(e)})
-
-    return jsonify({'message': 'OK'})
+        print(f'Failed to post data: {e}')
+        return jsonify({'Error': f'Failed to post data {e}'})
 
 @app.route('/context_broker/solution', methods=['GET', 'POST'])
 def get_solution():
-
     global fiware_node
-    solution = fiware_node.get_inference()
-
-    return jsonify(solution)
-
+    solution = fiware_node.get_inference(default_values.TIMEOUT_FIWARE_SOLUTION)
+    try:
+        sol=json.loads(solution)
+        return jsonify(sol)
+    except:
+        return jsonify(solution)

```

Comment:
Why do we need the `try-except` block if `get_inference()` always returns a `JSON`?

Reply:
Due to two reasons:
1) Due to the way solution is formatted when we receive the inference we can  not load it using json.loads().
2) Due to the fact than when we receive a JSON with an error message from the FiwareNode, if we return it directly using jsonify(solution), the frontend layer is not able to read correctly that there has been an error and treats it as if the inference was received correctly.

Reply:
It has been refactored